src/Uno.UI/Microsoft/UI/Xaml/Controls/RadioMenuFlyoutItem/RadioMenuFlyoutItem.cs

```diff
@@ -0,0 +1,153 @@
+			if (m_isChecked)
+			{
+				if (m_groupName != null)
+				{
+					s_selectionMap.Remove(m_groupName);
+				}
+			}
+		}
+
+		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
+			if (property == IsCheckedProperty)
+			{
+				if (base.IsChecked != IsChecked)
+				{
+					m_isSafeUncheck = true;
+					base.IsChecked = IsChecked;
+					m_isSafeUncheck = false;
+					UpdateCheckedItemInGroup();
+				}
+				m_isChecked = IsChecked;
+			}
+			else if (property == GroupNameProperty)
+			{
+				m_groupName = GroupName;
+			}
+		}
+
+		private void OnInternalIsCheckedChanged(DependencyObject sender, DependencyProperty args)
+		{
+			if (!base.IsChecked)
+			{
+				if (m_isSafeUncheck)
+				{
+					// The uncheck is due to another radio button being checked -- that's all right.
+					IsChecked = false;
+				}
+				else
+				{
+					// The uncheck is due to user interaction -- not allowed.
+					base.IsChecked = true;
+				}
+			}
+			else if (!IsChecked)
+			{
+				IsChecked = true;
+				UpdateCheckedItemInGroup();
+			}
+		}
+
+		private void UpdateCheckedItemInGroup()
+		{
+			if (IsChecked)
+			{
+				var groupName = GroupName;
+				if (s_selectionMap.TryGetValue(groupName, out var previousCheckedItemWeak))
+				{
+					if (previousCheckedItemWeak.TryGetTarget(out var previousCheckedItem))
+					{
+						// Uncheck the previously checked item.
+						previousCheckedItem.IsChecked = false;
+					}
+				}
+
+				s_selectionMap[groupName] = new WeakReference<RadioMenuFlyoutItem>(this);
+			}
+		}
+
+		private static void OnAreCheckStatesEnabledPropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			if (args.NewValue is true)
+			{
+				if (sender is MenuFlyoutSubItem subMenu)
+				{
+					// Every time the submenu is loaded, see if it contains a checked RadioMenuFlyoutItem or not.
+					subMenu.Loaded += (object sender, RoutedEventArgs _) =>
+					{
+						bool isAnyItemChecked = false;
+						foreach (var item in subMenu.Items)
+						{
+							if (item is RadioMenuFlyoutItem radioItem)
+							{
+								isAnyItemChecked = isAnyItemChecked || radioItem.IsChecked;
+							}
+						}
+						VisualStateManager.GoToState(subMenu, isAnyItemChecked ? "Checked" : "Unchecked", false);
+					};
+					/*
+					subMenu.Loaded(
+					{
+						[subMenuWeak = winrt::make_weak(subMenu)](winrt::IInspectable const& sender, auto const&)
+						{
+							if (auto subMenu = subMenuWeak.get())
+							{
+								bool isAnyItemChecked = false;
+								for (auto const& item : subMenu.Items())
+								{
+									if (auto const& radioItem = item.try_as<winrt::RadioMenuFlyoutItem>())
+									{
+										isAnyItemChecked = isAnyItemChecked || radioItem.IsChecked();
+									}
+								}
+								winrt::VisualStateManager::GoToState(subMenu, isAnyItemChecked ? L"Checked" : L"Unchecked", false);
+							}
+						}
+					});
+					 */
```

Comment:
I'm not sure I'm porting this code correctly. I'll remove the C++ commented code once someone confirm the C# equivalent is correct, or if not, how to fix it.

Specifically talking about the `[subMenuWeak = winrt::make_weak(subMenu)]` part. I think it's related to closure variable capture or something like that.

Reply:
That's correct, it's a closure.